pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -430,6 +430,43 @@ impl InMemoryLayer {
         self.put_value_locked(&mut inner, key, lsn, buf, ctx).await
     }
 
+    pub(crate) async fn put_batch(
+        &self,
+        buf: Vec<u8>,
+        values: Vec<(CompactKey, Lsn, u64)>,
+        ctx: &RequestContext,
+    ) -> Result<()> {
+        let mut inner = self.inner.write().await;
+        self.assert_writable();
+
+        let base_off = {
+            inner
+                .file
+                .write_raw(
+                    &buf,
+                    &RequestContextBuilder::extend(ctx)
+                        .page_content_kind(PageContentKind::InMemoryLayer)
+                        .build(),
+                )
+                .await?
+        };
```

Comment:
This means we will be now skipping page cache for in-memory layers. We want to do that anyway, but still worth noting.

Reply:
Right, this goes along nicely with the cutover to l0_flush direct mode that we did this week.